packages/manager/apps/telecom/src/components/telecom/telephony/scheduler/scheduler.controller.js

```diff
@@ -70,8 +70,13 @@ export default /* @ngInject */ function TelephonySchedulerCtrl(
 
     return self.scheduler
       .getEvents({
-        'dateStart.from': start.format(),
-        'dateEnd.to': end.format(),
+        'dateStart.from': moment(start)
+          .subtract(1, 'm')
```

Comment:
why would you subtract 1 month from start ?

Reply:
In order to get the previous month, as an event which start on n-1 of current month won't be retrieve by api

Reply:
for example if you have an event last month from the 1st to the 3rd it will be returned but it does not belong in the [start, end] interval (it is in the [start-1M, start] interval
is it an issue or is it simply not displayed in the front ?

Reply:
Yes it is an issue:
- An event which starts on last month (so event.dateStart < start) , but ends the current month (event.endDate > start), won't be displayed.

Same logic with the end: if event.dateStart < end + event.dateEnd > end, it won't be displayed.
So here I'm increasing the range to get those kind of events, so they will be displayed.

Reply:
I think 2 request should be made and results merge with uniqueness, because some events may be set for weeks, even months (lockdown is coming guys...)

Reply:
In fact @julieni, two requests as you suggest won't be enough.
If we set an event which last for 6 months, (starting on 2020-12-01, ending on 2021-05-01), it won't be catched.

The solution is that API allows rule as is:
- someEvent.dateStart <= request.dateStart <= someEvent.dateEnd.

Or to increase range from start of last year, to end of current year (with a little possibility to not catch event with larger time range).

Reply:
@JeremyDec, I think it will, and even 1 request should be enough, as I rechecked parameters of https://api.ovh.com/console/#/telephony/%7BbillingAccount%7D/scheduler/%7BserviceName%7D/events#GET

- dateEnd.from = first day of the period we want to display
- dateStart.to = last day of the period we want to display

In other words, for period d1 to d2: dateEnd >= d1 && dateStart <= d2

Seems to work, but I just woke up ;)

Reply:
Just tested:
- dateEnd.from = 2021-02-01 and dateStart.to = 2021-02-28 shows my event from 2021-01-29 to 2021-02-07
- dateEnd.from = 2020-12-28 and dateStart.to = 2021-01-31 shows my event from 2021-01-29 to 2021-02-07 too